File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1;
        f_i = 0
        f_ip1 = 1
        i = 0
        for i in range(n):
            f_i, f_ip1 = f_ip1, f_i + f_ip1
        return f_ip1

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climb(int n, std::unordered_map<int, int> &memo){
        if (n <= 2){
            return n;
        }
        if (!memo.contains(n)){
            memo[n] = climb(n-1, memo) + climb(n-2, memo);
        } 
        return memo[n];
    }
    int climbStairs(int n) {
        std::unordered_map<int, int> memo;
        return climb(n, memo);
    }
};